Mono.Nat/BufferHelpers.cs

```diff
@@ -0,0 +1,52 @@
+﻿//
+// Authors:
+//   Ben Motmans <ben.motmans@gmail.com>
+//
+// Copyright (C) 2020 Alan McGovern
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+
+using System.Threading;
+
+namespace Mono.Nat
+{
+	static class BufferHelpers
+	{
+		const int Size = 8 * 1024;
+
+#if NETSTANDARD2_0
+		static byte[] CachedBuffer;
+
+		internal static byte[] Rent()
+			=> Interlocked.Exchange(ref CachedBuffer, null) ?? new byte[Size];
+
+		internal static byte[] Release(byte[] buffer)
+			=> Interlocked.Exchange(ref CachedBuffer, buffer);
+#else
+		internal static byte[] Rent()
+			=> System.Buffers.ArrayPool<byte>.Shared.Rent(Size);
+
+		internal static void Release(byte[] buffer)
+			=> System.Buffers.ArrayPool<byte>.Shared.Return(buffer);
+#endif
```

Comment:
NS2.0 uses a single cached buffer, NS2.1 and newer will use ArrayPool. This ifdef should be future proof!